File: frontend/types/monitoring.ts
export type HealthStatus = 'Healthy' | 'Warning' | 'Error';

export interface ModuleHealth {
  moduleId: string;
  name: string;
  status: HealthStatus;
  uptime: string;
  activeThreads: number;
}

export interface QueueJob {
  id: string;
  moduleId: string;
  type: string;
  status: 'active' | 'pending' | 'completed' | 'failed';
  submitTimeStr: string;
  durationMs: number;
}

export interface ErrorEvent {
  id: string;
  moduleId: string;
  type: string;
  severity: 'low' | 'medium' | 'high' | 'critical';
  message: string;
  timeStr: string;
}

export interface PerformanceMetric {
  label: string;
  throughput: number;
  latencyMs: number;
  successRate: number;
}

export interface CorpusStatistic {
  documentCount: number;
  authorCount: number;
  entityCount: number;
  reviewCount: number;
}

export interface MonitoringSnapshot {
  id: string;
  health: ModuleHealth[];
  metrics: {
    queriesExecuted: number;
    queriesToday: number;
    querySuccessRate: number;
    reviewsGenerated: number;
    reviewsToday: number;
    registeredUsers: number;
    activeUsers: number;
    refreshTokensActive: number;
    documentsProcessed: number;
  };
  charts: PerformanceMetric[];
  queue: QueueJob[];
  recentErrors: ErrorEvent[];
  corpus: CorpusStatistic;
}

File: backend/api/schemas/monitoring.py
from pydantic import BaseModel
from typing import List, Optional

class CorpusStatistic(BaseModel):
    documentCount: int
    authorCount: int
    entityCount: int
    reviewCount: int

class ModuleHealth(BaseModel):
    moduleId: str
    name: str
    status: str
    uptime: str
    activeThreads: int

class QueueJob(BaseModel):
    id: str
    moduleId: str
    type: str
    status: str
    submitTimeStr: str
    durationMs: int

class ErrorEvent(BaseModel):
    id: str
    moduleId: str
    type: str
    severity: str
    message: str
    timeStr: str

class PerformanceMetric(BaseModel):
    label: str
    throughput: float
    latencyMs: float
    successRate: float

class MonitoringSnapshot(BaseModel):
    id: str
    health: List[ModuleHealth]
    metrics: dict
    charts: List[PerformanceMetric]
    queue: List[QueueJob]
    recentErrors: List[ErrorEvent]
    corpus: CorpusStatistic
